File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        hashmap = {}
        for n in nums:
            if n not in hashmap:
                hashmap[n] = 1
            else:
                hashmap[n] += 1
        for key, val in hashmap.items():
            if val > 1:
                return key
        return -1

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> sup;
        for (auto n : nums) {
            if (sup.contains(n)) {
                return true;
            }
            sup.insert(n);
        }
        return false;
    }
};